picard/webservice.py

```diff
@@ -246,17 +297,32 @@ def url_port(url):
             return url.port(443)
         return url.port(80)
 
-    def _handle_reply(self, reply, request, handler, xml, refresh):
+    def _handle_reply(self, reply, wsrequest):
+        handler = wsrequest.handler
         error = int(reply.error())
         if error:
-            log.error("Network request error for %s: %s (QT code %d, HTTP code %s)",
-                      reply.request().url().toString(QUrl.RemoveUserInfo),
-                      reply.errorString(),
-                      error,
-                      repr(reply.attribute(QtNetwork.QNetworkRequest.HttpStatusCodeAttribute))
-                      )
-            if handler is not None:
+            code = int(reply.attribute(QtNetwork.QNetworkRequest.HttpStatusCodeAttribute))
+            errstr = reply.errorString()
+            url = reply.request().url().toString(QUrl.RemoveUserInfo)
+            log.error("Network request error for %s: %s (QT code %d, HTTP code %d)",
+                      url,errstr, error, code)
+
+            if (wsrequest.retries < TEMP_ERRORS_RETRIES
+                and (code == 503
+                     or code == 429
+                     # following line is a workaround for Picard-809
+                     or errstr.endswith("Service Temporarily Unavailable")
+                    )
+               ):
+                new_wsrequest = wsrequest.copy()
+                new_wsrequest.retries += 1
+                new_wsrequest.priority = False
+                new_wsrequest.important = False
+                log.debug("Retrying (#%d) %s", new_wsrequest.retries, url)
```

Comment:
This should print the url first, like other debug messages in this module do. That's also easier to read because the url is the constant part of the message and the number of retries varies.